tests/models/test_ModelUtils.py

```diff
@@ -573,6 +574,37 @@ def test_get_named_leaf_modules() -> None:
     assert utils.get_named_leaf_modules(sequential2) == {"0.0": linear1, "0.1": linear2}
 
 
+@pytest.mark.parametrize(
+    "strategy, expected_fn",
+    [
+        ("learn", "initialize_learnable_positional_embedding"),
+        ("sincos", "initialize_2d_sine_cosine_positional_embedding"),
+        ("skip", None),
+    ],
+)
+def test_initialize_positional_embedding(
+    strategy: str, expected_fn: Optional[str], mocker: MockerFixture
+) -> None:
+    if expected_fn is not None:
+        mock_fn = mocker.spy(utils, expected_fn)
+    pos_embedding = Parameter(torch.rand(1, 1, 64))
+    utils.initialize_positional_embedding(
+        pos_embedding=pos_embedding, strategy=strategy, num_prefix_tokens=1
+    )
+    if expected_fn is not None:
+        mock_fn.assert_called_once()
+
+
+def test_initialize_learnable_positional_embedding() -> None:
+    pos_embedding = Parameter(torch.ones(1, 1, 64))
+    orig_pos_embedding = pos_embedding.clone()
+    utils.initialize_learnable_positional_embedding(pos_embedding)
+    # Embedding must be learnable.
+    assert pos_embedding.requires_grad
+    # Embedding must have changed.
+    assert torch.any(pos_embedding != orig_pos_embedding)
```

Comment:
This could fail randomly if the rng only gives out 0. Thus I would seed it.
Furthermore, I would rather check that e.g. at least 90% have changed (given a fixed seed).

Reply:
It would only fail if all values were randomly 0 which should be impossible. Let's keep the test as is.

Reply:
Ok, but if it fails because all values are randomly 0, you're gonna hear a big "I told you so!" ;)

Feel free to resolve.